agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h

```diff
@@ -552,6 +556,14 @@ class backhaul_manager : public btl::transport_socket_thread {
     std::vector<sApMetricsQuery> m_ap_metric_query;
     std::vector<sApMetricsResponse> m_ap_metric_response;
 
+    struct sChannelSelectionResponse {
+        sMacAddr radio_mac;
+        uint8_t response_code;
+    };
+
+    std::vector<sMacAddr> m_channel_selection_request;
+    std::vector<sChannelSelectionResponse> m_channel_selection_response;
```

Comment:
We should also store the MID, and check it when collecting responses.
Therefore, I think it would be nicer like this:
```suggestion
   struct sExpectedChannelSelection {
       uint16_t mid;
       std::vector<sMacAddr> requested;
       std::vector<sChannelSelectionResponse> responses;
   };
   sExpectedChannelSelection m_expected_channel_selection;
```

Reply:
Fixed.